### PR TITLE
docs(NODE-6516): add documentation for MongoOperationTimeoutError and fix internal references

### PR DESCRIPTION
**Pull request by nbbeeken**
#### What is changing?

- docs for MongoOperationTimeoutError

#### What is the motivation for this change?

Clarity on the error subject

### Double check the following

- [ ] Ran `npm run check:lint` script
- [ ] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [ ] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: etc/notes/errors.md [hunk 1]
```diff
@@ -111,7 +111,9 @@ This class should **never** be directly instantiated.
 
 ### `MongoOperationTimeoutError`
 
-- TODO(NODE-6491): Add MongoOperationTimeoutError documentation
+The `MongoOperationTimeoutError` class represents an error that occurs when an operation could not be completed within the specified `timeoutMS`.
+It is generated by the driver in support of the "client side operation timeout" feature so inherits from `MongoDriverError`.
```
Comment (dariakp) [inline etc/notes/errors.md:115]:
```suggestion
It is generated by the driver in support of the "client side operation timeout" feature and inherits from `MongoDriverError`.
```

File: etc/notes/errors.md [hunk 2]
```diff
@@ -111,7 +111,9 @@ This class should **never** be directly instantiated.
...
+When `timeoutMS` is enabled `MongoServerErrors` relating to `MaxTimeExpired` errors will be converted to `MongoOperationTimeoutError`
```
Comment (dariakp) [inline etc/notes/errors.md:116]:
```suggestion
When `timeoutMS` is enabled `MongoServerErrors` relating to `MaxTimeExpired` errors will be converted to `MongoOperationTimeoutError`.
```

File: test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts [hunk 3]
```diff
@@ -28,7 +28,6 @@ import {
 import { type FailPoint, makeMultiBatchWrite, measureDuration } from '../../tools/utils';
 import { filterForCommands } from '../shared';
 
-// TODO(NODE-5824): Implement CSOT prose tests
```
Comment (dariakp) [inline test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts:31]: There is a comment that says
`    'TODO(DRIVERS-2965): see modified test in unified-csot-node-specs', // Skipping for both tailable awaitData and tailable non-awaitData cursors
`
in client_side_operations_timeout.spec.test.ts - can we remove or update it in this PR?

**Review by dariakp — CHANGES_REQUESTED** [review]

File: etc/notes/errors.md (hunk 1 above)
Comment (nbbeeken) [reply etc/notes/errors.md]: fixed

File: etc/notes/errors.md (hunk 2 above)
Comment (nbbeeken) [reply etc/notes/errors.md]: fixed

File: test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts (hunk 3 above)
Comment (nbbeeken) [reply test/integration/client-side-operations-timeout/client_side_operations_timeout.prose.test.ts:31]: removed

**Review by dariakp — APPROVED** [review]

**Pull request merged**